components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -353,7 +355,17 @@ private Response getUnauthorizedResponseObject() {
 
     private URI getResourceLocationURI(CreateResourceDTO response) throws URISyntaxException {
 
-        return new URI(RESOURCE_PATH + "/" + response.getResourceId());
+        try {
+            String locationUri = ServiceURLBuilder.create().addPath(UMAConstants.REGISTERED_RESOURCE_PATH, response
+                    .getResourceId()).build().getAbsolutePublicURL();
+            return new URI(locationUri);
+        } catch (URLBuilderException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred while building the resource location URI with tenant qualified URL.", e);
+            }
+            // Fallback to legacy approach during error scenarios to maintain backward compatibility.
+            return new URI(RESOURCE_PATH + "/" + response.getResourceId());
```

Comment:
Why do we need to do this?

Reply:
added it as a fall back mechanism to maintain backward compatibility.. shall I remove it?

Reply:
I don't think we need to fall back to the legacy approach in the exception handling path. This might lead to an inconsistent behaviour in the case of the tenanat qualified mode, for a legitimate exception scenario.

We have used UMAConstants.REGISTERED_RESOURCE_PATH in the new impl, where as in the old impl we had used RESOURCE_PATH. Are these both the same?

Reply:
Fixed fall-back logic.

Earlier RESOURCE_PATH was not a constant but a variable to assign the value of IdentityUtil.getServerURL(REGISTERED_RESOURCE_PATH, true, true);
[1] https://github.com/wso2-extensions/identity-oauth-uma/blob/592b963f29ffcca7d0b6f16de791a80b8e5b6ef1/components/org.wso2.carbon.identity.oauth.uma.common/src/main/java/org/wso2/carbon/identity/oauth/uma/common/UMAConstants.java#L29